numpy/compat/_inspect.py

```diff
@@ -60,6 +61,35 @@ def iscode(object):
     """
     return isinstance(object, types.CodeType)
 
+def cleandoc(doc):
+    """Clean up indentation from docstrings.
+
+    Any whitespace that can be uniformly removed from the second line
+    onwards is removed."""
```

Comment:
Sure, cpython isn't always the best for documentation. If inspect imports are no longer a problem we could simplify things in the future. IIRC, @rkern had the concern.